analytics/tests/test_views.py

```diff
@@ -702,17 +702,23 @@ def test_activate_or_deactivate_realm(self) -> None:
         self.assertEqual(result.status_code, 302)
         self.assertEqual(result["Location"], "/login/")
 
-        self.login('iago')
+    def test_change_subdomain(self) -> None:
+        cordelia = self.example_user('cordelia')
+        lear_realm = get_realm('lear')
+        self.login_user(cordelia)
+        current_subdomain = lear_realm.string_id
 
-        with mock.patch("analytics.views.do_deactivate_realm") as m:
-            result = self.client_post("/activity/support", {"realm_id": f"{lear_realm.id}", "status": "deactivated"})
-            m.assert_called_once_with(lear_realm, self.example_user("iago"))
-            self.assert_in_success_response(["lear deactivated"], result)
+        result = self.client_post("/activity/support", {"realm_id": f"{lear_realm.id}",
+                                                        "change_subdomain": "new_name"})
+        self.assertEqual(result.status_code, 302)
+        self.assertEqual(result["Location"], "/login/")
+        self.login('iago')
 
-        with mock.patch("analytics.views.do_send_realm_reactivation_email") as m:
-            result = self.client_post("/activity/support", {"realm_id": f"{lear_realm.id}", "status": "active"})
-            m.assert_called_once_with(lear_realm)
-            self.assert_in_success_response(["Realm reactivation email sent to admins of lear"], result)
+        result = self.client_post("/activity/support", {"realm_id": f"{lear_realm.id}", "change_subdomain": "new_name"})
+        lear_realm = get_realm('new_name')
+        self.assert_in_success_response(["Subdomain of realm successfully changed new_name"], result)
+        self.assertEqual(lear_realm.string_id, "new_name")
+        self.assertNotEqual(lear_realm.string_id, current_subdomain)
```

Comment:
We don't need this check. Since we fetch lear realm by `lear_realm = get_realm('lear')`